src/main/java/seedu/address/logic/commands/DeleteLogCommand.java

```diff
@@ -0,0 +1,332 @@
+         * {@code CommandResult} object.
+         *
+         * @throws CommandException if invalid data state encountered.
+         */
+        public CommandResult applyDelete(Model model) throws CommandException {
+
+            CommandResult result;
+
+            if (isForOnePerson && !isForDeletingAllLogs) {
+                // case 1: delete specific log of specific person
+                result = this.deleteSpecificPersonLog(model);
+
+            } else if (isForOnePerson && isForDeletingAllLogs) {
+                // case 2: delete all logs of specific person
+                result = this.deleteAllLogsOfPerson(model);
+
+            } else if (!isForOnePerson && isForDeletingAllLogs) {
+                // case 3: delete all logs all persons
+                result = this.deleteAllLogs(model);
+
+            } else {
+                throw new CommandException(DeleteLogCommandParser.MESSAGE_INVALID_FORMAT);
+            }
+            return result;
+        }
+
+        /**
+         * Deletes all {@code Logs} objects of all persons in the address book.
+         **/
+        public CommandResult deleteAllLogs(Model model) {
+
+            // sanity check
+            assert (this.isForDeletingAllLogs && !this.isForOnePerson
+                    && this.personIndex == null && this.logIndex == null);
+
+            // get all persons and delete all
+            List<Person> allPersonsList = model.getAddressBook().getPersonList();
+            for (Person person : allPersonsList) {
+                Person editedPerson = copyPersonButWithEmptyLogs(person);
+                model.setPerson(person, editedPerson);
+            }
+            return new CommandResult(MESSAGE_DELETE_LOG_SUCCESS);
+        }
+
+        /**
+         * Carries out deletion logic for deletion of all {@code Log} objects of a specified person.
+         *
+         * @throws CommandException if person specified is not in address book.
+         */
+        public CommandResult deleteAllLogsOfPerson(Model model) throws CommandException {
+
+            // sanity check
+            assert (this.isForDeletingAllLogs && this.isForOnePerson
+                    && this.personIndex != null && this.logIndex == null);
+
+            // set new person as empty
+            // get list of persons from model
+            List<Person> lastShownList = model.getFilteredPersonList();
+
+            // get person and modify
+            if (this.personIndex.getZeroBased() >= lastShownList.size()) {
+                throw new CommandException(Messages.MESSAGE_INVALID_PERSON_DISPLAYED_INDEX);
+            }
+            Person personToEdit = lastShownList.get(this.personIndex.getZeroBased());
+            Person deletedLogsPerson = copyPersonButWithEmptyLogs(personToEdit);
+
+            // add to address book
+            model.setPerson(personToEdit, deletedLogsPerson);
+            model.updateFilteredPersonList(PREDICATE_SHOW_ALL_PERSONS);
```

Comment:
Minor nitpick, but do we want to updateFilteredPersonList so that it displays the edited friends instead of all friends?